verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -123,6 +124,28 @@ void IndexingFactsTreeExtractor::Visit(const verible::SyntaxTreeNode& node) {
       ExtractNetDeclaration(node);
       break;
     }
+    case NodeEnum::kPreprocessorDefine: {
+      ExtractMacroDefinition(node);
+      break;
+    }
+    case NodeEnum::kMacroCall: {
+      ExtractMacroCall(node);
+      break;
+    }
+    // Special case to handle tags in leaves.
+    // e.g leaves tagged with MacroIdentifier.
+    case NodeEnum::kUnqualifiedId: {
+      const verible::SyntaxTreeLeaf* leaf =
+          ABSL_DIE_IF_NULL(AutoUnwrapIdentifier(node));
+
+      // Leaves tagged with MacroIdentifier.
+      if (leaf->get().token_enum() == verilog_tokentype::MacroIdentifier) {
+        ExtractMacroReference(*leaf);
+        break;
+      }
```

Comment:
does this cover cases like:  parameter int x = TEN?

Reply:
I check it in CST and this should be covered by this part of code.

Reply:
You have to examine the CST structure `--printtree` to confirm.

I would however recommend that for macros, you use a implement a `::Visit(const SyntaxTreeLeaf&)` to simplify this.  Macro identifiers are neither qualified nor unqualified as they may appear anywhere.  This might actually cover `` `BAR`` in  `` `FOO(`BAR) `` as well.